ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,248 @@
+                    filter_vals=self.imagesets_used, condition_func=need_file, cache=self.cache)
+
+        # sequester data for this specific run    
+        cache = self.cache
+        cache.ensure_clean_subpath('data/temp')
+        cache.ensure_subpath_exists('data/temp')
+
+        data_dir = cache.path / 'data'
+
+        cli_spinner("Copying data into temp folder...", copy_data_locally,
+            source_dir=data_dir, dest_dir=self.temp_dir, condition_func=need_file)
+        
+        self.image_ids = image_ids
+        self.filter_metadata = filter_metadata
+    
+    @cli_spinner_wrapper("Writing out metadata locally...")
+    def write_metadata(self):
+        """Writes out a metadata file in JSON format
+
+        Args:
+            name (str): the name of the dataset
+            comments (str): comments or notes supplied by the user regarding the
+                dataset produced by this tool
+            training_type (str): the training type selected by the user
+            image_ids (list): a list of image IDs that ended up in the final
+                dataset (either dev or test)
+            filters (dict): a dictionary representing filter metadata
+            transforms (dict): a dictionary representing transform metadata
+            out_dir (Path, optional): Defaults to Path.cwd().
+        """
+        dataset_path = self.dataset_path / self.dataset_name
+        metadata_filepath = dataset_path / 'metadata.json'
+
+        metadata = {}
+        metadata["name"] = self.dataset_name
+        metadata["date_created"] = datetime.utcnow().isoformat() + "Z"
+        metadata["created_by"] = self.created_by
+        metadata["comments"] = self.comments
+        metadata["training_type"] = self.plugin_name
+        metadata["image_ids"] = self.image_ids
+        metadata["filters"] = self.filter_metadata
+        with open(metadata_filepath, 'w') as outfile:
+            json.dump(metadata, outfile) 
+
+    @cli_spinner_wrapper("Writing out dataset locally...")
+    def write_dataset(self, obj_list: list):
+        """Main driver for this file
+        
+        Args:
+            obj_list (list): objects that will be transformed into usable dataset
+            test_percent (float): percent of data that'll be test data
+            num_folds (int): number of folds to write
+            out_dir (Path): directory to write this dataset to
+            custom_dataset_name (str): name of the dataset's containing folder
+        """
+        dataset_path = self.dataset_path / self.dataset_name
+        print(dataset_path)
+        self.delete_dir(dataset_path) # Only deletes the contents already inside the directory
+
+        test_subset, dev_subset = self.split_data(obj_list)
+
+        # Test subset.
+        self.write_related_data(test_subset, dataset_path / 'test')
+
+        dev_path = dataset_path / 'splits'
+
+        # standard_path = dev_path / 'standard'
```

Comment:
We need to actually decide if we intend to keep using `standard` and `complete` or not. Our datasets are fractured right now due to this commenting-out leaking into master, let's not let that propagate. 

`standard` is where the kfolds are stored. If we intend to make kfold creation optional, lets do that. 

I think it is best to leave kfold as required, so that all our datasets have the ABILITY to be used for that type of validation if needed. That way, if we have some dataset that produces kickass models, we can go back and do kfold cross validation with it to double check ourselves and present more robust results.

Especailly since kfold functionality is already baked in, we should use it I think

Reply:
About kfold being required, the reason I originally commented it out is that it increases creation time and also makes the datasets way larger (I have very limited hard drive space so this is annoyingly a big concern for me). Even if it's not off by default, I do think it should at least be optional. Tbh, it seems to me like a lot of well-respected papers don't even do k-fold if the test set is large enough -- and we have the ability to generate unlimited data.

Reply:
making it optional is fine - we just need to start including a "kfold" field in the metadata so downstream training code can check for it and conduct training differently. the issue right now is that the pipeline was originally planned to always include the `standard` and `complete` directories with their corresponding `train`, `validation`, and `test` subdirectories. Plugins assumed this by default and currently have no way of knowing if a given dataset has `standard,` `complete`, both,  or neither.